server/src/main/java/com/linecorp/centraldogma/server/storage/encryption/RepositoryEncryptionStorage.java

```diff
@@ -578,6 +623,98 @@ private static byte[] repoCurrentWdekDbKey(String projectName, String repoName)
         return ("wdeks/" + projectName + '/' + repoName + "/current").getBytes(StandardCharsets.UTF_8);
     }
 
+    CompletableFuture<Void> rewrapAllWdeks() {
+        final List<WrappedDekDetails> allWdeks = wdeks();
+        if (allWdeks.isEmpty()) {
+            logger.info("No WDEKs to rewrap");
+            return CompletableFuture.completedFuture(null);
+        }
+
+        logger.info("Starting to rewrap WDEKs...");
+
+        final List<CompletableFuture<WrappedDekDetails>> rewrapFutures = new ArrayList<>();
+        for (WrappedDekDetails wdekDetails : allWdeks) {
+            final String oldKekId = wdekDetails.kekId();
+            logger.info("Re-wrapping WDEK for {}/{} version {} from KEK {} to {}",
+                       wdekDetails.projectName(), wdekDetails.repoName(),
+                       wdekDetails.dekVersion(), oldKekId, kekId);
+
+            final CompletableFuture<WrappedDekDetails> rewrapFuture =
+                    keyWrapper.rewrap(wdekDetails.wrappedDek(), oldKekId, kekId)
+                              .handle((wrappedDek, cause) -> {
+                                  if (cause != null) {
+                                      logger.warn("Failed to rewrap WDEK for {}",
+                                                  projectRepoVersion(wdekDetails.projectName(),
+                                                                     wdekDetails.repoName(),
+                                                                     wdekDetails.dekVersion()), cause);
+                                      return null;
+                                  }
+                                  if (oldKekId.equals(kekId) && wdekDetails.wrappedDek().equals(wrappedDek)) {
+                                      logger.info("WDEK for {} is already wrapped with the target KEK {}. " +
+                                                  "Skipping rewrap.",
+                                                  projectRepoVersion(wdekDetails.projectName(),
+                                                                     wdekDetails.repoName(),
+                                                                     wdekDetails.dekVersion()),
+                                                  kekId);
+                                      return null;
+                                  }
+
+                                  return new WrappedDekDetails(wrappedDek,
+                                                               wdekDetails.dekVersion(),
```

Comment:
Question) I may be misunderstanding, but given that the active dek is queried by version, is there no need to increment the version for the newly wrapped dek?

Reply:
>  I may be misunderstanding, but given that the active dek is queried by version, is there no need to increment the version for the newly wrapped dek?

No, it's not. 😉 
Because the dek isn't changed, the version is also not changed. It's just rewraping the dek with another kek. 😉

Reply:
Because KeK is compromised, we don't have to recreate DEKs. We can simpley rewrap the DEKs with the new KEK.
When DEKs are compromised, we need to recreate the DEKs. It will be handled in this line:
https://github.com/line/centraldogma/blob/9bc952d5200ed76a950fd2847c350b3e850e9dce/server/src/main/java/com/linecorp/centraldogma/server/internal/api/sysadmin/KeyManagementService.java#L70-L73